### PR TITLE
FEAT:CAP-40- Create a Standings Page

### PR DESCRIPTION
**Pull request by rooshnie23**
## Description

**JIRA Ticket:** [CAP-40](https://mfslcapstone.atlassian.net/browse/CAP-40)

**Summary:**
Added a new page that could be found "/standings", where there is the league standings page. It is a table that is sorted by league points and with the ranking system. The row gets highlighted whenever the mouse is hovering over it. Additionally, the current user's team is also highlighted when you get on the page.

**Screenshots/Examples:**
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-functional change which does not add or remove functionality)

## Checklist

- [X] My branch and PR name follow the [naming conventions](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4358145/Development+Naming+Conventions) the team has established.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.
- [X] Any dependent changes have been merged and published in downstream modules. (If not, this should be a draft PR)
- [X] I have [rebased](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4456479/How+To%3A+Rebase+Branch+Onto+Main) my branch onto main before opening this PR, and resolved any merge conflicts.
- [X] I have ensured that new and existing integrations work with my changes.

[CAP-40]: https://mfslcapstone.atlassian.net/browse/CAP-40?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/StandingsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    createData('Team 12', 40, 20, 10, 85, 26),
+    createData('Team 5', 42, 17, 9, 91, 28),
+    createData('Team 9', 43, 16, 8, 94, 29),
+];
+
+const currentUserTeam = 'Team 1';
+
+function StandingsTable() {
+    // Sort the rows based on league points
+    const sortedRows = initialRows.slice().sort((a, b) => b.leaguePts - a.leaguePts);
```
Comment (abhijeetpraveen) [inline src/components/StandingsTable.tsx:39]: I think it's better if we turn the sortedRows into a prop that can be accepted by this StandingsTable component. When the user lands on the actual StandingsPage, we can have a useEffect that based on the username, fetches the user's team name. Then we propagate that string down as a prop to the StandingsTable component in order to highlight the logged in user's team.

**Review by abhijeetpraveen — CHANGES_REQUESTED** [review]

File: src/components/StandingsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    createData('Team 2', 19, 32, 24, 37, 16),
+    createData('Team 3', 30, 23, 22, 51, 17),
+    createData('Team 4', 27, 21, 27, 45, 18),
+    createData('Team 11', 39, 18, 13, 82, 25),
+    createData('Team 12', 40, 20, 10, 85, 26),
+    createData('Team 5', 42, 17, 9, 91, 28),
+    createData('Team 9', 43, 16, 8, 94, 29),
+];
+
+const currentUserTeam = 'Team 1';
```
Comment (abhijeetpraveen) [inline src/components/StandingsTable.tsx:35]: I think it's better if we turn the currentUserTeam into a prop that can be accepted by this StandingsTable component. When the user lands on the actual StandingsPage, we can have a useEffect that based on the username, fetches the user's team name. Then we propagate that string down as a prop to the StandingsTable component in order to highlight the logged in user's team.

**Review by abhijeetpraveen — CHANGES_REQUESTED** [review]

File: src/components/StandingsTable.tsx (hunk 2 above)
Comment (rooshnie23) [reply src/components/StandingsTable.tsx]: Yes, thank you for the suggestion

**Comment by abhijeetpraveen** [comment]
image

Why are two different teams being highlighted. Shouldn't we only be seeing `Team 1` being highlighted. Are we hardcoding the last row to be highlighted as well?

**Comment by rooshnie23** [comment]
Team 1 is currently highlighted for the presentation. I proceeded to do the same for the props too. Would you like me to remove it? So, the second is highlighted when the user is hovering over the table. My mouse can't appear on screenshot, but it is hovering over team 11.

**Review by abhijeetpraveen — APPROVED** [review]

**Comment by abhijeetpraveen** [comment]
> Team 1 is currently highlighted for the presentation. I proceeded to do the same for the props too. Would you like me to remove it? So, the second is highlighted when the user is hovering over the table. My mouse can't appear on screenshot, but it is hovering over team 11. If the mouse is not hovering over it, it will be the same color has the others. I add this feature for the future where the user will able to click on the team and this is help for them.

Perfect, I see the hovering code now. That's amazing. You can merge it now.

**Pull request merged**